utils/helpers.py

```diff
@@ -102,7 +103,40 @@ def is_omariyya(case: dict,
     cond5 = n_siblings < 2
     return all([cond1, cond2, cond3, cond4, cond5])
 
+def is_akdariyya(case: dict) -> bool:
+    """
+    Check if this is an akdariyya problem
+    :param case:
+    :return:
+    """
+    cond1 = 'husband' in case
+    cond2 = ('sister' in case and 'brother' not in case) or ('paternal_halfsister' in case and 'paternal_halfbrother' not in case)
+    cond3 = 'mother' in case
+    invalidators = ['daughter', 'daughter_x2', 'daughter_of_son', 'daughter_of_son_x2']
+    cond4 = all([inh not in case for inh in invalidators])
+    return all([cond1, cond2, cond3, cond4])
+
 
 def nCr(n, r):
     f = math.factorial
     return f(n) / f(r) / f(n - r)
+
+def calculate_remainder(case: dict) -> dict:
```

Comment:
do you think it may make sense to rename this method such that we know it is specifically for the grandfather case?

Reply:
yeah done